src/overlays/actors/ovl_En_Niw_Girl/z_en_niw_girl.c

```diff
@@ -21,21 +31,226 @@ const ActorInit En_Niw_Girl_InitVars = {
+                            this->unk_27C, 0);
+    Math_SmoothScaleMaxF(&this->unk_27C, 5000.0f, 30.0f, 150.0f);
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+
+    //*Only allow Link to talk to her when she is playing the jumping animation
+    if ((this->jumpTimer == 0) || (func_8008F080(globalCtx) != 0)) {
+        this->jumpTimer = 60;
+        this->actionFunc = EnNiwGirl_Talk;
+    }
+}
+
+void EnNiwGirl_Talk(EnNiwGirl* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06009C78, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06009C78), 0, -10.0f);
+    this->actor.flags |= 1;
+    this->actor.textId = 0x7000; // I can't catch that Cucco!
+    if ((gSaveContext.eventChkInf[8] & 1) && (this->unk_27A == 0)) {
+        this->actor.textId = 0x70EA; // That scary-looking man took off out of here really fast !
+    }
+    switch (func_8008F080(globalCtx) - 1) {
+        case KEATON_MASK:
+            this->actor.textId = 0x7118; // Oh wow! It's Keaton! Hi, Keaton !
+            break;
+        case SPOOKY_MASK:
+            this->actor.textId =
+                0x7119; // How spooky! There's a boy in Kakariko Village who was looking for a mask like that!
+            break;
+        case BUNNY_HOOD:
+
+        case ZORA_MASK:
+        case GERUDO_MASK:
+            this->actor.textId = 0x711A; // Hmm... well... don't you have any other masks ?
+            break;
+        case SKULL_MASK:
+        case GORON_MASK:
+        case MASK_OF_TRUTH:
+            this->actor.textId = 0x711B; // What a strange mask!  Hee hah haa !
+            break;
+    }
+    this->unk_270 = 6;
+    this->actionFunc = func_80AB94D0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw_Girl/func_80AB918C.s")
+void func_80AB94D0(EnNiwGirl* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_8010BDBC(&globalCtx->msgCtx)) {
+        this->attachedActor->unk_2E8 = 0;
+    }
+    Math_SmoothDownscaleMaxF(&this->actor.speedXZ, 0.8f, 0.2f);
+    if (func_8002F194(&this->actor, globalCtx)) {
+        if (this->actor.textId == 0x70EA) { // That scary-looking man took off out of here really fast!
+            this->unk_27A = 1;
+            return;
+        }
+    } else {
+        if ((this->jumpTimer == 0) && !func_8010BDBC(&globalCtx->msgCtx)) {
+            this->jumpTimer = Math_Rand_ZeroFloat(100.0f) + 250.0f;
+            this->actionFunc = EnNiwGirl_PlayJumpAnimation;
+        } else {
+            func_8002F2CC(&this->actor, globalCtx, 100.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw_Girl/func_80AB9210.s")
+void EnNiwGirl_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnNiwGirl* this = THIS;
+    EnNiwGirlActionFunc tempActionFunc;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw_Girl/func_80AB93C0.s")
+    Actor_SetScale(&this->actor, 0.013f);
```

Comment:
Can thisx be used consistently in this file? (ie either not used at all or used everywhere instead of this->actor)
```suggestion
    Actor_SetScale(thisx, 0.013f);
```

Reply:
I should note that I meant consistently in functions where it's already available, not in every single function.